victim-loss-identifier/src/utils/utils.ts

```diff
@@ -0,0 +1,189 @@
+
+export function getChainBlockTime(chainId: number): number {
+  switch (chainId) {
+    case 10: // Optimism
+      return 2;
+    case 56: // BNB Smart Chain
+      return 3;
+    case 137: // Polygon
+      return 2;
+    case 250: // Fantom
+      return 2;
+    case 42161: // Arbitrum
+      return 1;
+    case 43114: // Avalance
+      return 2;
+    default: // Ethereum
+      return 12;
+  }
+}
+
+export function getBlocksInTimePeriodForChainId(timePeriodInSecs: number, chainId: number): number {
+  const chainBlockTime = getChainBlockTime(chainId);
+
+  return Math.floor(timePeriodInSecs / chainBlockTime);
+}
+
+export function cleanObject(object: { [key: string]: ScammerInfo } | { [key: string]: VictimInfo }) {
+  // Sort and delete 1/4 of the keys
+  const sortedKeys = Object.keys(object).sort((a, b) => {
+    const latestTimestampA = object[a].mostRecentActivityByBlockNumber;
+    const latestTimestampB = object[b].mostRecentActivityByBlockNumber;
+    return latestTimestampA - latestTimestampB;
+  });
+
+  // Ensure that at least one key will be deleted in the theoretical edge case where Math.floor(sortedKeys.length / 4) equals 0
+  const indexToDelete = Math.max(1, Math.floor(sortedKeys.length / 4));
+
+  for (let i = 0; i < indexToDelete; i++) {
+    delete object[sortedKeys[i]];
+  }
+}
+
+export function extractFalsePositiveDataAndUpdateState(
+  scammerAddress: string,
+  scammersCurrentlyMonitored: { [key: string]: ScammerInfo },
+  victims: { [key: string]: VictimInfo }
+) {
+  const fpData: FpTransaction[] = [];
+  const fpVictims: string[] = [];
+  const fpScammer = scammersCurrentlyMonitored[scammerAddress];
+
+  for (const victimAddress of Object.keys(fpScammer.victims!)) {
+    const victimInfo = fpScammer.victims![victimAddress];
+    const scammedBy = victimInfo.scammedBy;
+
+    const hasBeenAlerted = scammedBy[scammerAddress].hasBeenAlerted;
+    if (!hasBeenAlerted) continue;
+
+    const transactions = scammedBy[scammerAddress].transactions;
+
+    Object.keys(transactions).forEach((txHash) => {
+      const transaction = transactions[txHash];
+      const extractedTransaction: FpTransaction = {
+        txHash,
+        nfts: [],
+      };
+
+      if (transaction.erc721) {
+        // Iterate through ERC721 tokens
+        Object.keys(transaction.erc721).forEach((tokenAddress) => {
+          const erc721TokenInfo = transaction.erc721![tokenAddress];
+          const tokenIds = erc721TokenInfo.tokenIds;
+
+          if (victims[victimAddress].totalUsdValueAcrossAllErc721Tokens) {
+            victims[victimAddress].totalUsdValueAcrossAllErc721Tokens! -= erc721TokenInfo.tokenTotalUsdValue!;
+          }
+          // Format ERC721 data as "tokenId,tokenAddress"
+          tokenIds.forEach((tokenId) => {
+            extractedTransaction.nfts.push(`${tokenId},${tokenAddress}`);
+          });
+        });
+      }
+
+      if (transaction.erc1155) {
```

Comment:
Is this check to be more "future proof"? As in to already have in mind that there will be cases were a victim could be scammed of their ERC1155s? _Or_, is the bot supposed to handle the case of ERC1155 already?

If it is the latter, then the functions `addVictimInfoToVictims` and `increaseStolenUsdAmounts` in `fraud.nft.order.processing.ts` need to be updated to properly handle ERC1155 exploit instances.

Reply:
If it is the former, perhaps adding the logic to ERC20 transactions should be added.

WDYT?

Reply:
@RCantu92, it's the former, but ERC20s are not expected to generate a Label at all, I'd say, what do you think?

Reply:
E.g. in the Ice Phishing we're only labelling NFTs and I think that make sense due to ERC20s being totally fungible